lstests/core/service.py

```diff
@@ -0,0 +1,1526 @@
+            cnt = min(error_count, 10)
+            msg += f"\nReporting first {cnt} {helper.plural('exception', cnt)}:"
+            for i in range(cnt):
+                msg += f"\n[{i}] {channel[i]}"
+            self.error(msg)
+            if not dont_raise:
+                raise BlockchainError(msg)
+
+    def set_producers(self, producers:list=None, **call_kwargs):
+        if producers is None: producers = self.producers
+        prod_keys = []
+        for p in sorted(producers):
+            prod_keys.append({"producer_name": p, "block_signing_key": PRODUCER_KEY})
+        actions = [{"account": "eosio",
+                    "action": "setprods",
+                    "permissions": [{"actor": "eosio", "permission": "active"}],
+                    "data": { "schedule": prod_keys}}]
+        return self.push_actions(actions=actions, header="set producers", **call_kwargs)
+
+# --------------- auxiliary -------------------------------------------------------------------------------------------
+
+    def make_wasm_name(self, contract):
+        """Make a wasm filename given the contract name."""
+        return os.path.join(self.cdir, contract, contract + ".wasm")
+
+
+    def make_abi_name(self, contract):
+        """Make an abi filename given the contract name."""
+        return os.path.join(self.cdir, contract, contract + ".abi")
+
+
+    @staticmethod
+    def make_defproducer_name(num):
+        def base26_to_int(s: str):
+            res = 0
+            for c in s:
+                res *= 26
+                res += ord(c) - ord('a')
+            return res
+
+        def int_to_base26(x: int):
+            res = ""
+            while True:
+                q, r = divmod(x, 26)
+                res = chr(ord('a') + r) + res
+                x = q
+                if q == 0:
+                    break
+            return res
+        # 8031810176 = 26 ** 7 is integer for "baaaaaaa" in base26
+        return "defproducer" + string.ascii_lowercase[num] if num < 26 else "defpr" + int_to_base26(8031810176 + num)[1:]
+
+# --------------- check and verify ------------------------------------------------------------------------------------
+
+    def verify(self,
+               transaction_id,
+               verify_key=None,
+               verify_async=False,
+               dont_raise=False,
+               retry=None,
+               sleep=None,
+               level=None,
+               retry_level=None,
+               error_level=None,
+               buffer=None):
+        verify_async = helper.override(self.verify_async, verify_async, self.cla.verify_async)
+        retry = helper.override(self.verify_retry, retry, self.cla.verify_retry)
+        sleep = helper.override(self.verify_sleep, sleep, self.cla.verify_sleep)
+        verified = False
+        for __ in range(retry + 1):
+            if self.verify_transaction(transaction_id=transaction_id, verify_key=verify_key, level=retry_level, buffer=buffer):
```

Comment:
Async support appears to break down here.  While `def verify_transaction()` supports `**kwargs` and `def call()` supports the `verify_async` keyword argument, the `verify_async` flag calculated by `helper.override` is not provided to this call to `verify_transaction()`.

Reply:
Thank you for the observation. Though the original asynchronous verification functions well, I agree that there is room for improvement. To make the code logically clearer, I have refactored the verification part, with two updates:

1. It is now possible for the user to invoke asynchronous verification from two APIs:
    (1) `call()`
    (2) `verify()`

They can do so by passing `verify_async=True`. Previously it is only possible with `call`, not with `verify`. 

2. With either `call()` or `verify()` the overriding hierarchy of argument values is now four-fold:

default value << cluster-level argument << command-line-level argument << function-level argument

An example of the cluster-level argument is

```python
clus = Cluster(service=some_serv, verify_async=True)
```

An example of the function-level argument is

```python
clus.verify(transaction_id=some_id, verify_key="irreversible", verify_async=True)
```